### PR TITLE
[Merged by Bors] - feat(Algebra/Order/Hom): `Order(Add)MonoidIso`

### PR DESCRIPTION
**Pull request by pechersky**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by YaelDillies** [comment]
Thanks!

maintainer merge

File: Mathlib/Algebra/Order/Hom/Monoid.lean [hunk 1]
```diff
@@ -66,7 +71,8 @@ structure.
 `OrderAddMonoidHom` is also used for ordered group homomorphisms.
 
 When possible, instead of parametrizing results over `(f : α →+o β)`,
-you should parametrize over `(F : Type*) [OrderAddMonoidHomClass F α β] (f : F)`.
+you should parametrize over
+`(F : Type*) [FunLike F M N] [MonoidHomClass F M N] [OrderHomClass F M N] (f : F)`.
 
 When you extend this structure, make sure to extend `OrderAddMonoidHomClass`. -/
```
Comment (b-mehta) [inline Mathlib/Algebra/Order/Hom/Monoid.lean:77]: Looks like this needs to change too, like the above?

File: Mathlib/Algebra/Order/Hom/Monoid.lean [hunk 2]
```diff
@@ -446,6 +512,172 @@ end OrderedCommGroup
...
+
+@[to_additive (attr := simp)]
+theorem toOrderIso_eq_coe (f : α ≃*o β) : f.toOrderIso = f :=
+  rfl
+
+variable (f)
+
+@[to_additive]
+protected lemma strictMono : StrictMono f :=
+  strictMono_of_le_iff_le (fun _ _ ↦ (map_le_map_iff _).symm)
```
Comment (b-mehta) [inline Mathlib/Algebra/Order/Hom/Monoid.lean:654]: I think these parens are unneeded?

**Review by b-mehta — APPROVED** [review]

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**